internal/controller/policyserver_controller_deployment.go

```diff
@@ -220,28 +223,133 @@ func (r *PolicyServerReconciler) updatePolicyServerDeployment(policyServer *poli
+		if len(r.OtelCertificateSecret) > 0 {
+			policyServerDeployment.Spec.Template.Spec.Volumes = append(policyServerDeployment.Spec.Template.Spec.Volumes, corev1.Volume{
+				Name: otelCertificateVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: r.OtelCertificateSecret,
+					},
+				},
+			})
+			admissionContainer.VolumeMounts = append(admissionContainer.VolumeMounts, corev1.VolumeMount{
+				Name:      otelCertificateVolumeName,
+				ReadOnly:  true,
+				MountPath: certificatePath,
+			})
+		}
+		clientCertificatePath := filepath.Dir(os.Getenv("OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE"))
+		if len(r.OtelClientCertificateSecret) > 0 {
+			policyServerDeployment.Spec.Template.Spec.Volumes = append(policyServerDeployment.Spec.Template.Spec.Volumes, corev1.Volume{
+				Name: otelClientCertificateVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: r.OtelClientCertificateSecret,
+					},
+				},
+			})
+			admissionContainer.VolumeMounts = append(admissionContainer.VolumeMounts, corev1.VolumeMount{
+				Name:      otelClientCertificateVolumeName,
+				ReadOnly:  true,
+				MountPath: clientCertificatePath,
+			})
+		}
+		// As the controller is sending data to remote otel collector, we need
+		// to replicate the env vars to the policy server deployment. Thus, it
+		// will be able to send data to the same collector.
+		otelEnvVarToReplicate := []string{
+			"OTEL_EXPORTER_OTLP_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_CLIENT_KEY",
+			"OTEL_EXPORTER_OTLP_COMPRESSION",
+			"OTEL_EXPORTER_OTLP_ENDPOINT",
+			"OTEL_EXPORTER_OTLP_HEADERS",
+			"OTEL_EXPORTER_OTLP_INSECURE",
+			"OTEL_EXPORTER_OTLP_METRICS_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_METRICS_CLIENT_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_METRICS_CLIENT_KEY",
+			"OTEL_EXPORTER_OTLP_METRICS_COMPRESSION",
+			"OTEL_EXPORTER_OTLP_METRICS_DEFAULT_HISTOGRAM_AGGREGATION",
+			"OTEL_EXPORTER_OTLP_METRICS_ENDPOINT",
+			"OTEL_EXPORTER_OTLP_METRICS_HEADERS",
+			"OTEL_EXPORTER_OTLP_METRICS_INSECURE",
+			"OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE",
+			"OTEL_EXPORTER_OTLP_METRICS_TIMEOUT",
+			"OTEL_EXPORTER_OTLP_TIMEOUT",
+			"OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_TRACES_CLIENT_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_TRACES_CLIENT_KEY",
+			"OTEL_EXPORTER_OTLP_TRACES_COMPRESSION",
+			"OTEL_EXPORTER_OTLP_TRACES_ENDPOINT",
+			"OTEL_EXPORTER_OTLP_TRACES_HEADERS",
+			"OTEL_EXPORTER_OTLP_TRACES_INSECURE",
+			"OTEL_EXPORTER_OTLP_TRACES_TIMEOUT",
+		}
+		for _, envVar := range otelEnvVarToReplicate {
+			if value := os.Getenv(envVar); value != "" {
+				envvar := corev1.EnvVar{Name: envVar, Value: value}
+				if index := envVarsContainVariable(admissionContainer.Env, envVar); index >= 0 {
+					admissionContainer.Env[index] = envvar
+				} else {
+					admissionContainer.Env = append(admissionContainer.Env, envvar)
```

Comment:
Yes, I was thinking about that as well. Another option is to add some validation in the webhook to prevent the users from deploying a policy server with a configuration that can cause issues. Therefore, we can at least tell them why the controller cannot deploy the policy server. Instead of allow they to deploy and notice that something is not working as they expected afterwards.